src/index.js

```diff
@@ -0,0 +1,197 @@
+};
+
+
+const findLatitudeAndLongitude = () => {
+  let latitude, longitude;
+  return axios
+    .get("http://127.0.0.1:5000/location", {
+      params: {
+        q: state.cityName,
+        format: "json",
+      },
+    })
+    .then((response) => {
+      latitude = response.data[0].lat;
+      longitude = response.data[0].lon;
+      console.log("success in findLatitudeAndLongitude!", latitude, longitude);
+      return {
+        lat: latitude, 
+        lon: longitude, 
+      };
+    })
+    .catch((error) => {
+      console.log("error in findLatitudeAndLongitude!");
+      throw error;
+    });
+};
+
+const getTemperature = () => {
+
+  return findLatitudeAndLongitude()
+    .then((coordinates) => {
+      return findTemperature(coordinates);
+    })
+    .then((temp) => {
+      const fahrenheit = 1.8 * (temp - 273) + 32;
+      state.tempCount = Math.round(fahrenheit);
+      refreshUI();
+      updateTempColorAndLandscape();
+    })
+    .catch((error) => {
+      console.error("Error in getTemperature!");
+      state.tempValue.textContent = "Enter a valid city"
+    });
+
+// wave 5
+};
+const updateSky = () => {
+  const skySelect = document.getElementById('skySelect').value;
+  const skyContainer = document.getElementById('sky');
+
+  skyContainer.textContent = "";
+
+  let sky;
+  let skyBackgroundClass;
+
+  switch (skySelect) {
+    case 'sunny':
+      sky = '☀️ ☀️ ☀️ ☀️ ☀️ ☀️';
+      skyBackgroundClass = 'sunny';
+      break;
+    case 'cloudy':
+      sky = '☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️';
+      skyBackgroundClass = 'cloudy';
+      break;
+    case 'rainy':
+      sky = '🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧';
+      skyBackgroundClass = 'rainy';
+      break;
+    case 'snowy':
+      sky = '🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨';
+      skyBackgroundClass = 'snowy';
+      break;
+ 
+  }
+
+  const skyContent = document.createElement('div');
+  skyContent.textContent = sky;
+  skyContainer.appendChild(skyContent);
+  skyContainer.className = skyBackgroundClass;
+};
+
+
+const loadControls = () => {
+  state.tempValue = document.getElementById("tempValue");
+};
+
+const registerEventHandlers = () => {
+  loadControls();
+
+  const decreaseTemp = document.getElementById("decreaseTempControl");
+  decreaseTemp.addEventListener("click", updateDecreaseTempCount);
+
+  const increaseTemp = document.getElementById("increaseTempControl");
+  increaseTemp.addEventListener("click", updateIncreaseTempCount);
+  
+  const skySelect = document.getElementById('skySelect');
+  skySelect.addEventListener('change', updateSky);
+  
+  const cityNameReset = document.getElementById("cityNameReset");
+  cityNameReset.addEventListener("click", resetCityNameInput);
+  cityNameInput.addEventListener("input", updateCityNameInput);
+
+  const currentTempButton = document.getElementById("currentTempButton");
+  currentTempButton.addEventListener("click", getTemperature);
+  
+  updateTempColorAndLandscape();
+};
+
+document.addEventListener('DOMContentLoaded', registerEventHandlers);
```

Comment:
Nice work ensuring that your document was loaded because adding all your functionality!